src/app/checklist/app-updates/app-updates.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material';
+import { SwUpdate } from '@angular/service-worker';
+import {
+  fromEvent as observableFromEvent,
+  interval,
+  merge as observableMerge,
+  Observable,
+  of as observableOf
+} from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AppUpdatesService {
+  constructor(public snackbar: MatSnackBar, private serviceWorkerUpdates: SwUpdate) {}
+
+  checkUpdatesAuto() {
+    this.serviceWorkerUpdates.activated.subscribe(() => {
+      this.snackbar.open('App updated', 'OK', { duration: 2000 });
+    });
+
+    this.serviceWorkerUpdates.available.subscribe(() => {
```

Comment:
I'd be in favor of refactoring this piece of code as nested subscription are not ideal. We can use one of the flattening operations to get rid of this. Here's my suggestion:
```
this.serviceWorkerUpdates.available
  .pipe(
    switchMap(() => {
      const updateSnackbar = this.snackbar.open('Update available!', 'Update Now');
      return updateSnackbar.onAction().pipe(tap(() => updateSnackbar.dismiss()));
    })
  )
  .subscribe(() => window.location.reload());
```

I have used `switchMap` here because it's safe to unsubscribe from the previous snackbar's action stream as there's a new one ceated for each available update.

I have also removed the `duration` because we want to always enforce new versions, right? If the snackbar would go away and the user keeps the app in a tab alive then they would never get an update. So I think it's a good idea for the snackbar to be very prominent and also not disappear.

What do you think @KwintenP?